pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -190,6 +241,29 @@ func (i *Controller) handleUpdatePod(old interface{}, newObj interface{}) {
 		utilruntime.HandleError(err)
 		return
 	}
+
+	oldPodIp := old.(*k8sv1.Pod).Status.PodIP
+	updatedPodIp := newObj.(*k8sv1.Pod).Status.PodIP
+	podHostIP := newObj.(*k8sv1.Pod).Status.HostIP
+	// Pod events that are skipped during addEvent are handled here when they are assigned an ipaddress.
+	if updatedPodIp != "" && oldPodIp != updatedPodIp {
+		klog.V(4).Infof("In handleUpdatePod, pod %s is now assigned %s address, enqueing", old.(*k8sv1.Pod).Name, updatedPodIp)
+		i.enqueueObject(newObj, i.podWorkqueue)
+		return
+	}
+
+	// When the POD is getting terminated, sometimes we get pod update event with podIp removed.
+	if oldPodIp != "" && updatedPodIp == "" {
+		klog.V(4).Infof("Pod %s with ip %s is being terminated", old.(*k8sv1.Pod).Name, oldPodIp)
+		i.handleRemovedPod(old)
```

Comment:
`handleRemovedPod` expects the object's key. I'd suggest to change `handleRemovedPod` to take a `*k8sv1.Pod` factor out `extractPodFromKey` function.